src/main.rs

```diff
@@ -7,46 +7,87 @@ mod terminal;
 mod traits;
 mod ui;
 
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
 
 use anyhow::Result;
 use application::Application;
 use libc::{EXIT_FAILURE, EXIT_SUCCESS};
-use log::{info, LevelFilter};
+use log::{info, warn, LevelFilter};
 use terminal::TerminalWrapper;
 
-fn init_logging() -> log2::Handle {
-    let log_dir = if let Ok(_dir) = std::env::var("XDG_RUNTIME_DIR") {
-        // store log in the current directory for convenience
-        // we use XDG_RUNTIME_DIR to detect the fact that we are running on desktop linux
-        PathBuf::from("./")
+const EVE_MONITOR_LOG_DIR_EVE: &str = "/persist/monitor/log";
+const EVE_MONITOR_LOG_DIR: &str = "./persist/monitor/log";
+
+fn get_base_log_dir() -> &'static str {
+    // we use XDG_RUNTIME_DIR to detect the fact that we are running on desktop linux
+    // FIXME: is there a better way?
+    if let Ok(_dir) = std::env::var("XDG_RUNTIME_DIR") {
+        EVE_MONITOR_LOG_DIR
     } else {
-        // get current data and time and use it as a log file name
-        let now = chrono::Local::now();
-        // make /persist/monitor-<date>-<time>/ folder path and create he folder
-        let log_dir = PathBuf::from(format!(
-            "/persist/monitor-{}",
-            now.format("%Y-%m-%d-%H-%M-%S")
-        ));
-        std::fs::create_dir_all(&log_dir).expect("Failed to create log directory");
-        // set EVE_MONITOR_LOG_DIR to the created folder
-        std::env::set_var("EVE_MONITOR_LOG_DIR", log_dir.to_string_lossy().to_string());
-
-        log_dir
-    };
+        EVE_MONITOR_LOG_DIR_EVE
+    }
+}
+
+fn remove_old_log_sessions<T: AsRef<Path>>(log_dir: T, rotate_count: usize) -> Result<()> {
```

Comment:
Why templating here?

Reply:
@uncleDecart how else would you define it? this is not dynamic dispatch so it is a template. no other options